src/test/lrsql/ops/query/reaction_test.clj

```diff
@@ -0,0 +1,217 @@
+           lrs
+           "reaction-bad-template"
+           (merge
+            tc/simple-reaction-ruleset
+            {:template
+             ;; Template with invalid path
+             {"actor"  {"mbox" {"$templatePath" ["x" "actor" "mbox"]}}
+              "verb"   {"id" "https://example.com/verbs/completed"}
+              "object" {"id"         "https://example.com/activities/a-and-b"
+                        "objectType" "Activity"}}})
+           true)]
+      (try
+        ;; Add statements
+        (doseq [s [tc/reaction-stmt-a
+                   tc/reaction-stmt-b]]
+          (Thread/sleep 100)
+          (lrsp/-store-statements lrs tc/auth-ident [s] []))
+        (is (= {:result
+                [{:reaction-id reaction-id
+                  :trigger-id  trigger-id
+                  :error       {:type "ReactionTemplateError",
+                                :message
+                                "No value found at [\"x\" \"actor\" \"mbox\"]"}}]}
+               (qr/query-statement-reactions
+                bk ds {:trigger-id trigger-id})))
+        (finally (component/stop sys'))))))
+
+(deftest query-statement-reactions-invalid-statement-error-test
+  ;; Turn off instrumentation for this one
+  ;; This will keep the instrumentation error from clobbering
+  (support/unstrument-lrsql)
+  (testing "Invalid statement output"
+    (let [sys        (support/test-system
+                      :conf-overrides
+                      {[:lrs :enable-reactions] false})
+          sys'       (component/start sys)
+          lrs        (-> sys' :lrs)
+          bk         (:backend lrs)
+          ds         (-> sys' :lrs :connection :conn-pool)
+          trigger-id (u/str->uuid (get tc/reaction-stmt-b "id"))
+          {reaction-id
+           :result}
+          (adp/-create-reaction
+           lrs
+           "reaction-invalid-statement"
+           (merge
+            tc/simple-reaction-ruleset
+            {:template
+             ;; Template with invalid statement
+             {"foo" {"$templatePath" ["a" "actor" "mbox"]}}})
+           true)]
+      (try
+        ;; Add statements
+        (doseq [s [tc/reaction-stmt-a
+                   tc/reaction-stmt-b]]
+          (Thread/sleep 100)
+          (lrsp/-store-statements lrs tc/auth-ident [s] []))
+        (is (= {:result
+                [{:reaction-id reaction-id
+                  :trigger-id  trigger-id
+                  :error
+                  {:type "ReactionInvalidStatementError",
+                   :message
+                   "Reaction Invalid Statement Error - Spec Error: #:statement{:foo \"mailto:bob@example.com\"} - failed: (contains? % :statement/actor) spec: :xapi-schema.spec/statement
+#:statement{:foo \"mailto:bob@example.com\"} - failed: (contains? % :statement/verb) spec: :xapi-schema.spec/statement
+#:statement{:foo \"mailto:bob@example.com\"} - failed: (contains? % :statement/object) spec: :xapi-schema.spec/statement
+"}}]}
```

Comment:
And _especially_ here

Reply:
f7c2a469c1f298382066540859134e25593c593c